cmd/Dockerfile

```diff
@@ -1,35 +1,37 @@
-FROM quay.io/centos/centos:stream9 as builder
+FROM --platform=$BUILDPLATFORM docker.io/golang:1.23 AS builder
```

Comment:
go ver is hardcoded here instead taking from go.mod
i believe we can live with it, just raising for awareness in case you think otherwsie

Reply:
This doesn't affect D/S (at least our as far as i understand), and also better to usually be considered implicit
it also require changing on more than one place, which isnt what we strive to do usually,
only rarely we need a different compiler than a language level imho

Reply:
Ok, let's use go.mod if this is standard practice in kubevirt-related projects. I updated PR to rollback to explicit go download.